api/v1alpha1/clusterextension_types.go

```diff
@@ -73,8 +73,41 @@ type ClusterExtensionSpec struct {
 	// the bundle may contain resources that are cluster-scoped or that are
 	// installed in a different namespace. This namespace is expected to exist.
 	InstallNamespace string `json:"installNamespace"`
+
+	//+kubebuilder:Optional
+	// Preflight defines the strategy for the preflight checks (i.e. as of now, the CRD upgrade safety checks) to be applied or skipped when attempting to install the cluster extension.
+	Preflight PreflightConfig `json:"preflight,omitempty"`
+}
+
+// PreflightConfig holds the config for the preflight checks. Currently, this is implemented for the
+// CRDUpgradeSafety preflight check and can be extended in the future to add other preflight checks.
+type PreflightConfig struct {
+	CRDUpgradeSafety CRDUpgradeSafetyPreflightConfig `json:"crdUpgradeSafety,omitempty"`
+}
+
+// CRDUpgradeSafetyPreflightConfig is a custom struct that holds the necessary configuration values to
+// enable or disable the CRD Upgrade Safety validations. Currently, this holds Mode
+// that sets the CRD Upgrade Safety validations to either Enabled/Disabled.
+// By default, this is set to "Enabled".
+type CRDUpgradeSafetyPreflightConfig struct {
+	//+kubebuilder:validation:Enum:=Enabled;Disabled
+	//+kubebuilder:default:=Enabled
+	//+kubebuilder:Optional
+	CRDUpgradeSafetyMode CRDUpgradeSafetyMode `json:"crdUpgradeSafetyMode,omitempty"`
```

Comment:
I think for the field name I'd still call this mode otherwise you have to define it in yaml like:
```yaml
...
preflight:
  crdUpgradeSafety:
    crdUpgradeSafetyMode: Disabled
...
```

which, IMO, feels redundant as opposed to:
```yaml
...
preflight:
  crdUpgradeSafety:
    mode: Disabled
...
```

Reply:
Ah my bad, based on [this comment](https://github.com/operator-framework/operator-controller/pull/845#discussion_r1600201812), I had assumed that you were suggesting to change the field name in the struct and also the json tag to `Mode`. 
Just realized you were suggesting to change the field name in the struct alone. 

Fixed it now in [commit](https://github.com/operator-framework/operator-controller/pull/845/commits/fffc98a0a57f9d860aab3a2347a36bd8edfe28bd)